src/averageAges.js

```diff
@@ -53,6 +83,36 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  const mothers = people.filter(
+    mother => mother.sex === 'f' && people.some(
+      child => (onlyWithSon ? child.sex === 'm' : true)
+        && child.mother === mother.name
+    )
+  );
+
+  const children = people.filter(
+    child => (onlyWithSon ? child.sex === 'm' : true)
+      && mothers.some(mother => mother.name === child.mother)
+  )
+    .map(
+      child => {
+        const currentMother = mothers.find(
+          mother => mother.name === child.mother
+        );
+
+        child.bornOfMother = currentMother.born;
+
+        return child;
+      }
+    );
+
+  const totalAgeDiff = children.reduce(
+    (sum, child) => sum + child.born - child.bornOfMother, 0
+  );
+
+  const averageAgeDiff = totalAgeDiff / children.length;
+
+  return averageAgeDiff;
```

Comment:
Last 3 lines of code could be combined (in previous functions also)
return children.reduce(
    (sum, child) => sum + child.born - child.bornOfMother, 0
  ) / children.length;